### DISCUSSION TITLE
Component children changed event

### DISCUSSION THREAD

**Discussion by ahmafi** [GrapesJS/grapesjs#4494] [Q&A]
When I'm writing custom components, are there any events that trigger when another component or block gets dropped into my custom component?

In the [docs](https://grapesjs.com/docs/modules/Components.html#define-custom-component-type), I saw only `change:attributes` and `change:someprop`  events in the `init` method of model section. I tried `change:components` but it didn't do anything.

I want something like this:

```js
export const plugin = (editor) => {
  editor.DomComponents.addType('my-component', {
    model: {
      defaults: {
        ...
        components: [<div class="col"></div>, <div class="col"></div>],
        ...
      },
    },

    init() {
      this.on('change:components', () => {
        // a component or block dropped into "my-component"
        console.log('something happened');
      });
    },
  });
};
```
I would also like to know if there are any other events than `change:something` in the component `init` hook.

**Comment by artf** [comment]
Did you try `component:update:components`?

**Reply by ahmafi** [reply]
Isn't this on the editor level? If I do `editor.on('component:update:components', ...)`, then it will get called on every component, but I want to listen for changes only on this component that I'm defining.
Is there any event only for the current component?

**Reply by artf** [reply]
Yeah probably it's a bit confusing right now but this particular event (`component:update:*`) could be attached at the component level.
